### PR TITLE
Hint/feature/notify all users

### PR DESCRIPTION
**Pull request by dinjas**
Addresses #468.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -34,6 +34,9 @@ class User < ApplicationRecord
   has_many :blockouts, dependent: :destroy
   belongs_to :race, optional: true
   has_and_belongs_to_many :age_ranges
+  has_many :annoucements,
+           dependent:   :restrict_with_error,
```
Comment (dinjas) [inline app/models/user.rb:38]: @benjaminwood do you think this is the appropriate dependency?

Comment (benjaminwood) [reply app/models/user.rb:38]: Yeah, that's good.

Comment (dinjas) [reply app/models/user.rb:38]: Coolio, thanks for sanity check!

File: app/admin/announcements.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+ActiveAdmin.register Announcement do
+  config.sort_order = 'created_at_desc'
+
+  actions :index, :show, :new, :create
+  permit_params :message
```
Comment (benjaminwood) [inline app/admin/announcements.rb:7]:
```suggestion
  permit_params :message
```

If this is not defined it'll use the policy `permitted_attributes`

File: app/admin/announcements.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  before_create do |annoucement|
+    annoucement.author   = current_user
+    annoucement.user_ids = User.announceable.pluck(:id)
+  end
+
+  after_create(&:send_messages)
```
Comment (benjaminwood) [inline app/admin/announcements.rb:14]: Nice, I didn't know this was a thing.

File: app/models/announcement.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Announcement < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+
+  validates :author,
```
Comment (benjaminwood) [inline app/models/announcement.rb:6]: Do you think we should limit the `message` length to 160 characters?

Comment (dinjas) [reply app/models/announcement.rb:6]: We talked about that. Might not be a bad idea. The text field looks pretty big on the screen - they might be tempted to fill it up.

Comment (nvick) [reply app/models/announcement.rb:6]: +1 for limiting

Comment (dinjas) [reply app/models/announcement.rb:6]: Will add

File: app/admin/announcements.rb (hunk 2 above)
Comment (dinjas) [reply app/admin/announcements.rb:7]: this comment confuses me haha. is there a suggested change here?

Comment (benjaminwood) [reply app/admin/announcements.rb:7]: LOL. Sorry, I've updated it!

Comment (dinjas) [reply app/admin/announcements.rb:7]: lol, thanks!

File: app/models/announcement.rb (hunk 4 above)
Comment (dinjas) [reply app/models/announcement.rb:6]: put a lower limit of 5 chars on it. lemme know if you want it changed or dont want that at all :)

**Comment by dinjas** [comment]
I am deploying the latest to staging. I edited several staging users so their phone numbers are verified so this can be tested there. There were a couple of (key) users missing phone completely, so wasn't able to set them as verified (Ben and Sarah).

**Pull request merged**